src/sklearn_evaluation/plot/regression.py

```diff
@@ -113,14 +110,16 @@ def prediction_error(y_true, y_pred, model=None, ax=None):
     if ax is None:
         ax = plt.gca()
 
-    model = model or LinearRegression()
-    if not hasattr(model, "fit_intercept"):
-        raise TypeError(
-            '"fit_intercept" attribute not in model. ' "Cannot plot prediction error."
-        )
-    # best fit line
-    model.fit(y_true.reshape((-1, 1)), y_pred)
-    x = np.linspace(80, 230, 100)
+    model = LinearRegression()
+
+    if isinstance(y_true, pd.Series):
+        y_true = y_true.values
+    y_reshaped = y_true.reshape((-1, 1))
+
+    # it is necessary to fit the model with y_true and y_pred
+    # to get the best fit line representing the error trend
+    model.fit(y_reshaped, y_pred)
```

Comment:
On analysing the code I realised that `model.fit(y_reshaped, y_pred)` is used for fitting the best fit line over the y_pred values. So this doesn't seem an issue. However, the `model` parameter seems unnecessary and might confuse the user. Since we just need a linear best fit line, `LinearRegression()` should suffice and can be the default. Other regressors like `Lasso()` or `Ridge()` are generating similar best fit lines as `LinearRegression()`. But this will cause the method signature to change. Thoughts? @edublancas @idomic

Reply:
So `Lasso/Ridge/LinearRegression(y_reshaped, y_pred)` instead of `model.fit(y_reshaped, y_pred)`?

Reply:
can you look up in the literature? see if there are situations where a user might want to fit a Lasso or Ridge? If we find some justification we can leave the argument there

Reply:
I skimmed through some papers, didn't get exactly what we're looking for ( prediction error plot that also displays the `best fit` line). However below are some observations:
1. One common approach is to visualise the perpendicular distance of the error from the model's best fit line : 
<img width="455" alt="error_plot" src="https://user-images.githubusercontent.com/29096135/210548242-f8eefbde-f17e-4ace-b126-6e376a8b317a.png">

2. The closest that I could find was the 
[fit plot](https://www.sfu.ca/~mjbrydon/tutorials/BAinPy/09_regression.html#fit-plot) from `statsmodel`. However this doesn't visualise any `best fit` line, and shows the scatter plot only.

A few papers I referred to: (didn't find the prediction error plot with best fit line in these)
[On the interpretation of regression plots](https://sci-hub.se/https://www.tandfonline.com/doi/abs/10.1080/01621459.1994.10476459?journalCode=uasa20)
[Evaluating Random Forests for Survival Analysis using Prediction Error Curves](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4194196/)
[Modern approach to regression](https://d1wqtxts1xzle7.cloudfront.net/59391552/Sheather_-_A_Modern_Approach_to_Regression_with_R20190525-2191-a6v3gs-libre.pdf?1558803412=&response-content-disposition=inline%3B+filename%3DSpringer_Texts_in_Statistics.pdf&Expires=1672840294&Signature=KiJKPz0x~UUi~h3bpip6mdmIaWdCqs4HvEKPWXqVZkERnQazHQsY1H6dFpDfoDLLQ-9n~u-UnVY77wjsedqHaQ644queDSlVnId1BOWUJn9qf~vrKbK86MZ21vVIHRiQc8uq~i7nF-d8S4ZFr2FsIl6qbeDS9YJMqiKnJAAp5QSuIOpVQ9dH9LqJ7mfrPmk17iM87INLJkf6bULAEOEvoM7Ru7jDiBXV72Gyw4PWd8TMtxPxhlvzrYTSpT6dE-GGLsb8xvdzfQOYnhk1dqhBIL~xk5mv1MLnF12V0hFOXBpggiS4N1IIVq-eUJd4wmJPLt6fkZ6jtl2piBWjDWPlyg__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA)

Reply:
ok, it makes sense to remove the model argument then. from what I understand, its only purpose is to serve as a benchmark to compare model predictions to the 45-degree line. let's deprecate it and remove it in an upcoming release. https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-deprecated

Reply:
> ok, it makes sense to remove the model argument then. from what I understand, its only purpose is to serve as a benchmark to compare model predictions to the 45-degree line. let's deprecate it and remove it in an upcoming release. https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-deprecated

Yes that's right.
Added the deprecated directive.